src/authkit-callback-route.ts

```diff
@@ -96,7 +96,7 @@ export function handleAuth(options: HandleAuthOptions = {}) {
         await saveSession({ accessToken, refreshToken, user, impersonator }, request);
 
         if (onSuccess) {
-          await onSuccess({
+          const onSuccessResponse = await onSuccess({
```

Comment:
suggestion: I wonder if it makes sense to provide the default response as an argument to onSuccess? That way, you don't always need to create a new one from scratch.

This would also help make it more explicit that if `onSuccess` returns a `NextResponse`, it's going to take precedence over `returnPathname` if both are provided.